Source/Tools/LinuxEmulation/LinuxSyscalls/GdbServer.cpp

```diff
@@ -639,8 +661,41 @@ GdbServer::HandledPacketType GdbServer::CommandReadRegisters(const fextl::string
   // Pause up front
   SyscallHandler->TM.Pause();
   const FEX::HLE::ThreadStateObject* CurrentThread = FindThreadByTID(CurrentDebuggingThread);
+  const size_t NumGPR = Is64BitMode() ? FEXCore::Core::CPUState::NUM_GPRS : FEXCore::Core::CPUState::NUM_GPRS / 2;
+  const size_t GPRSize = Is64BitMode() ? sizeof(uint64_t) : sizeof(uint32_t);
+  const size_t NumXMM = Is64BitMode() ? FEXCore::Core::CPUState::NUM_XMMS : FEXCore::Core::CPUState::NUM_XMMS / 2;
+  const size_t XMMSize = Is64BitMode() ? sizeof(__uint128_t) * 2 : sizeof(__uint128_t);
+  fextl::string str;
   auto GDB = GenerateContextDefinition(CurrentThread);
-  return {encodeHex((unsigned char*)&GDB, sizeof(GDBContextDefinition)), HandledPacketType::TYPE_ACK};
+  for (size_t i = 0; i < NumGPR; ++i) {
+    str += appendHex(reinterpret_cast<const unsigned char*>(&GDB.gregs[i]), GPRSize);
+  }
+  str += appendHex(reinterpret_cast<const unsigned char*>(&GDB.rip), GPRSize);
+  str += appendHex(reinterpret_cast<const unsigned char*>(&GDB.eflags), sizeof(uint32_t));
+
+  str += appendHex(reinterpret_cast<const unsigned char*>(&GDB.cs), sizeof(uint32_t));
+  str += appendHex(reinterpret_cast<const unsigned char*>(&GDB.ss), sizeof(uint32_t));
+  str += appendHex(reinterpret_cast<const unsigned char*>(&GDB.ds), sizeof(uint32_t));
+  str += appendHex(reinterpret_cast<const unsigned char*>(&GDB.es), sizeof(uint32_t));
+  str += appendHex(reinterpret_cast<const unsigned char*>(&GDB.fs), sizeof(uint32_t));
+  str += appendHex(reinterpret_cast<const unsigned char*>(&GDB.gs), sizeof(uint32_t));
+  for (auto& mm : GDB.mm) {
+    str += appendHex(reinterpret_cast<const unsigned char*>(&mm), sizeof(X80Float));
+  }
+
+  str += appendHex(reinterpret_cast<const unsigned char*>(&GDB.fctrl), sizeof(uint32_t));
+  str += appendHex(reinterpret_cast<const unsigned char*>(&GDB.fstat), sizeof(uint32_t));
+  for (auto& dummy : GDB.dummies) {
+    str += appendHex(reinterpret_cast<const unsigned char*>(&dummy), sizeof(uint32_t));
+  }
```

Comment:
Actually can't this just be `str += appendHex(reinterpret_cast<const unsigned char*>(GDB.dummies), sizeof(GDB.dummies));`?

Reply:
Switched.